src/main/java/org/checkerframework/specimin/UnsolvedClass.java

```diff
@@ -204,6 +206,24 @@ public void updateFieldByType(String currentType, String correctType) {
     }
   }
 
+  @Override
+  public boolean equals(@Nullable Object other) {
+    if (!(other instanceof UnsolvedClass)) {
+      return false;
+    }
+    UnsolvedClass otherClass = (UnsolvedClass) other;
+    // This makes me nervous, because it's not clear that it
```

Comment:
An `UnsolvedClass` instance represents a synthetic class to be created by Specimin. And we don't create anonymous synthetic classes, so this case will not happen.

Reply:
Thanks for correcting this. I wrote a comment there instead to clarify why this is safe.